lib/cli/src/commands/run.rs

```diff
@@ -121,7 +120,7 @@ impl RunWithPathBuf {
 
             #[cfg(feature = "wasi")]
             {
-                let default = HashMap::default();
+                let default = indexmap::IndexMap::default();
```

Comment:
For the history books: can you 

* explain here in the PR what problem this solves
* Add comments in the code why `IndexMap` is used
* Open an issue to discuss how we want overlapping file system semantics to work

Reply:
See https://github.com/wasmerio/wasmer/issues/3492 - we need IndexMap to have a stable ordering for the [fs] mapping, otherwise overlapping filesystem mappings might not work. For example, in the `wasmer_toml`, we need to map `/lib`, then `/`, the order is important so that the mappings are mounted in the correct order. If we use `HashMap` here, then the order of the mounts are not preserved as they are written in the `wasmer.toml` file.

Reply:
@theduke done